### ISSUE TITLE
Crash report when allowing notifications during initial syncing

### ISSUE THREAD

**Issue by bluepuma77** [SamSchott/maestral#429]
Crash report when allowing notifications during initial syncing. Also showing sync issues.

**To Reproduce**
I installed Maestral on my Macbook M1, linked my Dropbox, it started syncing. I got a notification about notifications, so I clicked on it, System Preferences showed up with Notifications. I allowed notifications for Maestral. Got a crash report. Also there are sync issues.

**Expected behaviour**
No crash report.

**System:**

 - Maestral version: GUI v1.4.6, daemon v1.4.6
 - Python version: python 2.7.16, python3 3.8.2
 - OS: macOS Big Sur 11.4 
 - Desktop environment: default
 - PyQt version (for Linux GUI): "ModuleNotFoundError: No module named 'PyQt5'"

**Additional context**
```
Traceback (most recent call last):
File "Contents/Resources/app_packages/maestral/manager.py", line 557, in _handle_sync_thread_errors
File "Contents/Resources/app_packages/maestral/manager.py", line 535, in startup_worker
File "Contents/Resources/app_packages/maestral/sync.py", line 2776, in download_sync_cycle
File "Contents/Resources/app_packages/maestral/sync.py", line 2934, in apply_remote_changes
File "Contents/Resources/Support/Python/Resources/lib/python3.9/concurrent/futures/_base.py", line 608, in result_iterator
File "Contents/Resources/Support/Python/Resources/lib/python3.9/concurrent/futures/_base.py", line 445, in result
File "Contents/Resources/Support/Python/Resources/lib/python3.9/concurrent/futures/_base.py", line 390, in __get_result
File "Contents/Resources/Support/Python/Resources/lib/python3.9/concurrent/futures/thread.py", line 52, in run
File "Contents/Resources/app_packages/maestral/sync.py", line 3279, in _create_local_entry
File "Contents/Resources/app_packages/maestral/sync.py", line 3338, in _on_remote_file
File "Contents/Resources/app_packages/maestral/client.py", line 550, in download
FileNotFoundError: [Errno 2] No such file or directory
```

<img width="364" alt="Maestral-issues-screenshot" src="https://user-images.githubusercontent.com/5221187/126955384-cef3ef17-155f-4fc6-8f2d-ba161ab21343.png">

**Comment by SamSchott** [comment]
Oh, that is very strange. Just to confirm, are you seeing a hard crash (segfault) or is Maestral itself displaying an error message with the above traceback?

The exception from the traceback appears to happen just a after a file is downloaded, when trying to set its permission. Interestingly, it’s a `FileNotFoundError` for a file which was literally just created. Is there any other program that is modifying files or folders in the same directory? Regardless, it’s a race condition which we should handle better.

The sync issues likely have a similar cause: parent directories disappearing after we created them.

Could you post the log output here? If you are using the CLI, you can display it with `maestral log show -e`. Otherwise, the log files should be at `~/Library/Application Support/Logs/maestral`.

**Comment by bluepuma77** [comment]
@SamSchott I think it was a Mistral error message. Seemed like an integrated crash reporter, mentioned GitHub.

After the crash report the taskbar icon was first 'x', then '"'. So it seems the GUI did not crash.

I don't think there was any other app accessing the files. Except if macOS was doing something.

Strangely the crash report popped up the second I enabled maestral notifications in the System Preferences.

I found the logs at `~/Library/Logs/maestral`:
```
2021-07-26 09:53:17 sync INFO: Indexing 4110...
2021-07-26 09:53:17 sync INFO: Creating folders...
2021-07-26 09:53:17 sync INFO: Creating folder 1/4...
2021-07-26 09:53:20 sync INFO: Syncing ↓ 1/490
2021-07-26 09:53:22 sync INFO: Syncing ↓ 54/490
2021-07-26 09:53:24 sync INFO: Syncing ↓ 127/490
2021-07-26 09:53:26 sync INFO: Syncing ↓ 180/490
2021-07-26 09:53:28 sync INFO: Syncing ↓ 221/490
2021-07-26 09:53:31 sync INFO: Syncing ↓ 273/490
2021-07-26 09:53:33 sync INFO: Syncing ↓ 281/490
2021-07-26 09:53:40 sync INFO: Syncing ↓ 331/490
2021-07-26 09:53:42 sync INFO: Syncing ↓ 428/490
2021-07-26 09:53:44 sync INFO: Indexing 4610...
2021-07-26 09:53:44 sync INFO: Creating folders...
2021-07-26 09:53:44 sync INFO: Creating folder 1/1...
2021-07-26 09:53:58 sync INFO: Could not sync company_0513_3_extended_summary.pdf
Traceback (most recent call last):
  File "Contents/Resources/app_packages/maestral/client.py", line 152, in convert_api_errors
  File "Contents/Resources/app_packages/maestral/client.py", line 536, in download
FileNotFoundError: [Errno 2] No such file or directory: '/Users/user/Dropbox (Maestral)/.maestral.cache/tmp9c8n85n8'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "Contents/Resources/app_packages/maestral/sync.py", line 3279, in _create_local_entry
  File "Contents/Resources/app_packages/maestral/sync.py", line 3343, in _on_remote_file
  File "Contents/Resources/app_packages/maestral/sync.py", line 3338, in _on_remote_file
  File "Contents/Resources/app_packages/maestral/client.py", line 541, in download
  File "Contents/Resources/Support/Python/Resources/lib/python3.9/contextlib.py", line 135, in __exit__
  File "Contents/Resources/app_packages/maestral/client.py", line 169, in convert_api_errors
maestral.errors.NotFoundError: Could not sync file or folder. The given path does not exist.
2021-07-26 09:53:58 sync INFO: Could not sync company_0613_2_pitchdeck.pdf
Traceback (most recent call last):
  File "Contents/Resources/app_packages/maestral/client.py", line 152, in convert_api_errors
  File "Contents/Resources/app_packages/maestral/client.py", line 536, in download
FileNotFoundError: [Errno 2] No such file or directory: '/Users/user/Dropbox (Maestral)/.maestral.cache/tmp86lgr2f7'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "Contents/Resources/app_packages/maestral/sync.py", line 3279, in _create_local_entry
  File "Contents/Resources/app_packages/maestral/sync.py", line 3343, in _on_remote_file
  File "Contents/Resources/app_packages/maestral/sync.py", line 3338, in _on_remote_file
  File "Contents/Resources/app_packages/maestral/client.py", line 541, in download
  File "Contents/Resources/Support/Python/Resources/lib/python3.9/contextlib.py", line 135, in __exit__
  File "Contents/Resources/app_packages/maestral/client.py", line 169, in convert_api_errors
maestral.errors.NotFoundError: Could not sync file or folder. The given path does not exist.
2021-07-26 09:53:58 sync INFO: Could not sync company_0613_new_pitchdeck.pptx
Traceback (most recent call last):
  File "Contents/Resources/app_packages/maestral/client.py", line 152, in convert_api_errors
  File "Contents/Resources/app_packages/maestral/client.py", line 536, in download
FileNotFoundError: [Errno 2] No such file or directory: '/Users/user/Dropbox (Maestral)/.maestral.cache/tmpb44jja3f'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "Contents/Resources/app_packages/maestral/sync.py", line 3279, in _create_local_entry
  File "Contents/Resources/app_packages/maestral/sync.py", line 3343, in _on_remote_file
  File "Contents/Resources/app_packages/maestral/sync.py", line 3338, in _on_remote_file
  File "Contents/Resources/app_packages/maestral/client.py", line 541, in download
  File "Contents/Resources/Support/Python/Resources/lib/python3.9/contextlib.py", line 135, in __exit__
  File "Contents/Resources/app_packages/maestral/client.py", line 169, in convert_api_errors
maestral.errors.NotFoundError: Could not sync file or folder. The given path does not exist.
2021-07-26 09:55:27 manager ERROR: Unexpected error
Traceback (most recent call last):
  File "Contents/Resources/app_packages/maestral/manager.py", line 557, in _handle_sync_thread_errors
  File "Contents/Resources/app_packages/maestral/manager.py", line 535, in startup_worker
  File "Contents/Resources/app_packages/maestral/sync.py", line 2776, in download_sync_cycle
  File "Contents/Resources/app_packages/maestral/sync.py", line 2934, in apply_remote_changes
  File "Contents/Resources/Support/Python/Resources/lib/python3.9/concurrent/futures/_base.py", line 608, in result_iterator
  File "Contents/Resources/Support/Python/Resources/lib/python3.9/concurrent/futures/_base.py", line 445, in result
  File "Contents/Resources/Support/Python/Resources/lib/python3.9/concurrent/futures/_base.py", line 390, in __get_result
  File "Contents/Resources/Support/Python/Resources/lib/python3.9/concurrent/futures/thread.py", line 52, in run
  File "Contents/Resources/app_packages/maestral/sync.py", line 3279, in _create_local_entry
  File "Contents/Resources/app_packages/maestral/sync.py", line 3338, in _on_remote_file
  File "Contents/Resources/app_packages/maestral/client.py", line 550, in download
FileNotFoundError: [Errno 2] No such file or directory
2021-07-26 10:07:36 manager INFO: Retrying failed syncs...
2021-07-26 10:07:36 sync INFO: Syncing ↓ /company (chris)/old/company_0613_2_pitchdeck.pdf
2021-07-26 10:07:37 sync INFO: Syncing ↓ /company (chris)/old/company_0513_3_extended_summary.pdf
2021-07-26 10:07:39 sync INFO: Syncing ↓ /company (chris)/old/company_0613_new_pitchdeck.pptx
2021-07-26 10:07:40 sync INFO: Resuming indexing
2021-07-26 10:07:41 sync INFO: Indexing 4610...
2021-07-26 10:07:41 sync INFO: Creating folders...
2021-07-26 10:07:41 sync INFO: Creating folder 1/1...
2021-07-26 10:07:44 manager INFO: Rebuilding index...
2021-07-26 10:07:44 manager INFO: Shutting down threads...
2021-07-26 10:08:51 sync INFO: Syncing ↓ 1/453
2021-07-26 10:08:57 sync INFO: Syncing ↓ 18/453
2021-07-26 10:08:57 sync INFO: Sync aborted
2021-07-26 10:08:57 manager INFO: Paused
2021-07-26 10:08:57 sync INFO: Indexing remote Dropbox
```

**Comment by SamSchott** [comment]
Thanks for the log, this is already quite helpful. Maestral downloads any files to a cache directory `~/Dropbox (Maestral)/.maestral.cache/` and then moves them from this temporary location to the target location. It appears that either the cache dir or the temporaries files themselves get deleted just before or during the download. I have looked through the relevant code and cannot imagine what would cause this - the cache dir is only cleared once on startup.

Could run the sync again and this time explicitly observe what is happening in `~/Dropbox (Maestral)/.maestral.cache/`? You'll need to show hidden files and folders first (`CMD + SHIFT + .` shortcut in Finder).

I think (hope) that any relation with allowing  notifications is purely coincidental timing.

**Comment by bluepuma77** [comment]
After the crash I selected "Rebuild Index" and then it worked without issues.

Maybe this is the issue: I just found `Dropbox (Maestral)` and `Dropbox` in my home. I created a shortcut of the sync folder in Finder favourites and renamed it to `Dropbox`. I thought it would just rename the shortcut, but it seems it has renamed the original folder, too.

**Comment by SamSchott** [comment]
Indeed, renaming a folder in the Favourites bar renames the actual folder itself. This would explain why the cache directory "disappeared".

Typically when the root folder is moved, an error message will pop up and ask you to specify the new location. This message may be delayed during a download sync - and in your case it never appeared at all because the missing cache raised some unhandled exceptions. The latter should be fixed in 96d4307.

Note that you can actually change of the name of the root folder by creating a new empty folder and selecting in the preference pane.
